dvc/data/stage.py

```diff
@@ -247,6 +247,24 @@ def _get_staging(odb: "ObjectDB") -> "ObjectDB":
     return ReferenceObjectDB(fs, fs_path, state=state)
 
 
+def _load_untyped_dir_hash(odb, fs_path, hash_info, meta, staging):
+    from .tree import Tree
+
+    obj = Tree.load(odb, hash_info.as_raw())
+    obj.hash_info = hash_info
+    meta.nfiles = len(obj)
+
+    staging.add(obj.fs_path, obj.fs, obj.hash_info)
+    for key, (_, entry_hash_info) in obj.iteritems():
+        staging.add(
+            odb.fs.path.join(fs_path, *key),
+            odb.fs,
```

Comment:
we are not adding files from odb.fs, we are adding them from fs. Even though they happen to match here, it is not going to be always like that.
```suggestion
            fs.path.join(fs_path, *key),
            fs,
```